genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/queries/specifications/JpaCommandSpecs.java

```diff
@@ -114,5 +114,8 @@ private JpaCommandSpecs() {
                 CommandEntity_.id,
                 criterion
             );
+
+            return cb.and(predicate, cb.isNotEmpty(root.get(CommandEntity_.clusterCriteria)));
```

Comment:
Would this be more efficient if the order was swapped?

Reply:
As discussed offline this is unknown to me due to the layers the path would go through to generate the sql as well as if it's better to short circuit on a join clause vs an in row clause first. For now we'll try this and if we see a performance problem we can try switching it.